backend/logic/courses/courses.js

```diff
@@ -261,12 +261,104 @@ var updateCourse = async (course, newTitle, newTags) => {
   }
 };
 
+/**
+ * Assign a course to a curriculum
+ * @author Yufei Liu
+ * @param {string} curriculumType,{string} courseCode,{string} curriculum
+ * @returns true if successful
+ * @throws error if curriculumType, courseCode or curriculum dose not exist
+ *         course already assigned a curriculum
+ *
+ */
+var assignCourseToCurriculum = async (
+  curriculumType,
+  courseCode,
+  curriculum
+) => {
+  let conn = await mysql.getNewConnection();
+
+  const checkExistQuery = `SELECT *
+  FROM (
+      SELECT course_code, curriculum_name FROM curriculum_tech_comps
+      union all
+      SELECT course_code, curriculum_name FROM curriculum_complementaries
+      union all
+      SELECT course_code, curriculum_name FROM curriculum_core_classes
+  ) a
+  WHERE course_code = ? AND curriculum_name = ?;`;
+
+  // Check if the course has already been assigned, verify if curriculum type, course and curriculum are existed in DB
+  let ifAssigned;
+  let checkCourse;
+  let checkCurriculum;
+
+  try {
+    ifAssigned = await conn.query(checkExistQuery, [courseCode, curriculum]);
+    checkCourse = await conn.query(
+      "SELECT * FROM courses WHERE course_code = ?",
+      [courseCode]
+    );
+    checkCurriculum = await conn.query(
+      "SELECT * FROM curriculums WHERE curriculum_name = ?",
```

Comment:
```suggestion
      "SELECT COUNT(*) FROM curriculums WHERE curriculum_name = ?",
```

Reply:
The reason why I didn't use COUNT(*) is because it will return `RowDataPacket { 'COUNT(*)': 0 }`, then I need to convert it to a JSON, and read the `0` in the JSON, which would be too much. So I prefer to keep it like this, what do you think ?

Reply:
Hm yeah that thats an interesting problem. I had to try out a few things before I figure it out. In the query you can actually just do 
```
SELECT COUNT(*) AS count 
..
```
And then access it through 
```
result[0].count
```

Reply:
thanks! gonna fix it now